stacker/plan.py

```diff
@@ -150,8 +152,16 @@ def add(self, stack, run_func, requires=None):
             stack=stack,
             run_func=run_func,
             requires=requires,
+            tail=self.tail
         )
 
+    def poll(self):
+        stack_dict = self._poll_func(self.tail)
+        for step_name, step in self.list_pending():
+            if step_name in stack_dict:
```

Comment:
Not going to change this because 99% of the time, not all the stacks will exist in the event polls so this behaviour better suits the expected stack_dict